controllers/utils/etcd.go

```diff
@@ -33,6 +36,30 @@ func IsEtcdDisruptionAllowed(ctx context.Context, cl client.Client, log logr.Log
 		log.Info("Etcd disruption is allowed")
 		return true, nil
 	}
-	log.Info("Etcd disruption is not allowed")
+
+	// No disruptions allowed, so the only case we should remediate is that the node in question is already one of the disrupted ones
+	// The PDB doesn't disclose which node is disrupted
+	// So we have to check the etcd guard pods
+	podList := &corev1.PodList{}
+	if err := cl.List(ctx, podList, &client.ListOptions{
+		Namespace:     etcdNamespace,
+		LabelSelector: labels.SelectorFromSet(pdb.Labels),
```

Comment:
Which labels does the `etcd-guard-pdb` have? I was looking locally, and I haven't found any.

Reply:
Oh, good catch! Should be pdb.spec.selector. Done 🙂